teps/000X-trigger-crd.md

```diff
@@ -0,0 +1,351 @@
+<!-- /toc -->
+
+## Summary
+
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This proposal is to extract out Trigger definition from EventListener Spec into its own CRD 
+to resolve the problem of Multitenant EventListener.
+
+This is a TEP for the implementation of the Trigger CRD that were
+discussed in the following docs: 
+
+- [Trigger Listener CRDs](https://docs.google.com/document/d/1zUVrIbGZh2R9dawKQ9Hm1Cx3GevKIfOcRO3fFLdmBDc/edit#heading=h.tp9mko2koenr)
+- [Multitenant EventListener](https://docs.google.com/document/d/1NX0ExhPad6ixTM8AdU0b6Vc3MVD5hQ_vIrOs9dIXq-I/edit) 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this TEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a TEP within the wider Tekton community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Today, EventListener commonly are created for every namespace and handle TriggerBinding and
+TriggerTemplate in that same namespace. Creation of EventListener causes pod proliferation. For
+every namespace that requires handling of webhook, we need EventListener which in turns lead to
+this pod proliferation and causes excess resource consumption.
+
+### Goals
+
+<!--
+List the specific goals of the TEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+1. Reduction in resource cosumption due to EventListener.
+2. Ability to have single EventListener to cater to whole cluster.
+
+### Non-Goals
+
+<!--
+What is out of scope for this TEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+How to create Multitenant EventListener resource with appropriate permissions won't be part of this proposal.
+This will be handled by operator or admin of the cluster.
+
+## Requirements
+
+<!--
+List the requirements for this TEP.
+-->
+1. Add a new CRD Trigger which will be extracted from Trigger definition in EventListener Spec.
+2. Bind this Trigger CRD to EventListener.
+3. Ability to run Trigger resource in a different namespace from EventListener namespace.
+
```

Comment:
i think we need to be more clear about what the "requirements" section is for via the template, but the idea here is to explain constraints on the problem you are solving, e.g. (just making this up) it must be possible for one event listener to be able to create PipelineRuns in multiple namespaces

Reply:
How about?
1. Can create Tekton resources in a single namespace or across multiple namespaces if configured to do so.

2. When an EL recieves an event, it can process only some configured subset of triggers <- path/selector based

3. EL can still process triggers defined in the current form aka backwards compatibility.